### PR TITLE
Initial method-parsing HTTP endpoint

### PR DESCRIPTION
**Pull request by eddyashton**
This PR includes several changes to the inheritance structure of endpoints to create a parallel path for HTTP requests. This is still a work-in-progress, but I believe this should be sufficient to a) pass the tests, b) discuss the design, and c) move forward with related work (auth + full parsing + error reporting).

In short, we have `HTTPServerEndpoint` which does a similar job to `RPCEndpoint`. Both are derived from `Endpoint` via `TLSEndpoint` (via `HTTPEndpoint` and `FramedTLSEndpoint` respectively), and both parse some incoming message into an `RPCContext` to be processed by a targeted frontend. An important change from the previous version is that `Endpoint::send` takes the _body_ of a request, and the implementation of `send` wraps this with an appropriate header - size-prefixing or building a HTTP header - then actually transmits by calling `TLSEndpoint::send_raw` (or `send_buffered`, or...). For the `RPCClient` (used for the node-to-node join request), we have a `HTTPClientEndpoint` equivalent, and the key method is now `send_request(path, data)` since the HTTP case needs explicit context.

TODOs for next PRs:
- Parsing headers
- Extracting and verifying auth header
- Returning HTTP error codes where appropriate (execution errors will, for now, be a `200 OK` containing a JSON-RPC error, but parsing errors should fail at the HTTP level)
- Building fake params objects for bodyless `GET` requests

### CODE REVIEW

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

File: src/enclave/rpcendpoint.h [hunk 1]
```diff
@@ -2,35 +2,26 @@
...
 namespace enclave
 {
-#ifdef HTTP
-  using ServerEndpoint = HTTPEndpoint<http::ResponseHeaderEmitter>;
-#else
-  using ServerEndpoint = FramedTLSEndpoint;
-#endif
-
-  class RPCEndpoint : public ServerEndpoint
+  class RPCEndpoint : public FramedTLSEndpoint
```
Comment (jumaffre) [inline src/enclave/rpcendpoint.h]: `RPCEndpoint` is quite generic. Should this be `FramedJSONRPCEndpoint` instead?

Comment (eddyashton) [reply src/enclave/rpcendpoint.h]: Yes - I think a lot of the `RPC_` types should become `FramedJSONRPC_` or similar, but I'd like to save this for a separate PR.

**Pull request merged**